arm/proofs/ldst.ml

```diff
@@ -0,0 +1,159 @@
+(*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT-0
+ *)
```

Comment:
`ldst.ml` seems to contain sanity 'proof' tests for the new instructions, which is similar to what simulator does in the sense that it inspects the concrete output of the instructions. What about moving these proofs to simulator.ml, or even create a new simulate_ldst.ml & move these proofs to there & let simulator.ml load simulate_ldst.ml?
This can also help avoiding running these proofs everytime whenever `base.ml` is loaded; base.ml is supposed to contain lemmas and tactics that can be reused for general purposes, but it seems proofs in ldst.ml cannot be used that way.

Reply:
Thanks for the recommendation! I was having doubt myself if base.ml is the right place for it. I will update the PR.